pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -851,6 +849,31 @@ protected void handleCloseConsumer(CommandCloseConsumer closeConsumer) {
         }
     }
 
+    @Override
+    protected void handleGetLastMessageId(CommandGetLastMessageId getLastMessageId) {
+        checkArgument(state == State.Connected);
+
+        CompletableFuture<Consumer> consumerFuture = consumers.get(getLastMessageId.getConsumerId());
+
+        if (consumerFuture != null && consumerFuture.isDone() && !consumerFuture.isCompletedExceptionally()) {
+            Consumer consumer = consumerFuture.getNow(null);
+            long requestId = getLastMessageId.getRequestId();
+
+            Position position = consumer.getSubscription().getTopic().getLastMessageId();
+
+            log.info("[{}] [{}][{}] Get LastMessageId {}", remoteAddress,
+                consumer.getSubscription().getTopic().getName(), consumer.getSubscription().getName(), position);
+            MessageIdData messageId = MessageIdData.newBuilder()
```

Comment:
Missing partition, which i guess you get from the subscription. How does this deal with batch messages? What if the returned position is a batch message? The message id returned to the client wouldn't actually exist in the stream. Not sure how to deal with this.

Reply:
Thanks. Seems we got here only ledgerid and messageid. will noted in the client api that it only support non-partitioned topic.  partition and batch index here should be default value.

Reply:
It would be good to at least support partition id so that partitions can be compacted (this feature is needed for compaction). You can likely get it from the subscription or topic name. @merlimat could you weight in on this.

You probably can't get batch id because it's embedded in the message, so only the client has access. The only way to get this would be to implement the whole feature as sending the entire last message to the client, which likely isn't sensible. However, this does mean equality and comparisons may be messed up, as the returned last message id may not actually match the message id of a message in the topic.

Reply:
noticed this just now, will add partition here, if it is needed.

Reply:
Actually, maybe it would make more sense to add partition at the client side? From the broker point of view, a partition really just looks like another topic, it has it's own managed ledger. It's only from the client side that partition and batch id are important. (@merlimat @sijie what are your opinions on this)?

Reply:
@merlimat @sijie could you chime in on whether the partition id should be added on the server side or on the client side?

Reply:
I think it should be at client side, because currently "partition" related logic is at client side.